src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -141,7 +137,7 @@ public unsafe BitArray(bool[] values)
 
             if (Vector256.IsHardwareAccelerated)
             {
-                for (; (i + Vector256ByteCount) <= (uint)values.Length; i += Vector256ByteCount)
+                for (; (i + Vector256<byte>.Count) <= (uint)values.Length; i += (uint)Vector256<byte>.Count)
```

Comment:
```suggestion
                for (; (i) <= (uint)values.Length - Vector256<byte>.Count; i += (uint)Vector256<byte>.Count)
```
With the suggestion `i` can be used from a register directly, same for the length-count.
With the code as is `i` needs to be added first, then compared.

See the difference in [M1 and M2 from sharplab (silly) example](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8ovLrl5hcAbho1iAZiakGAYQYBvGg11NlxFAwCy5ABSwAZg2ABPDDAa4IHKGBhoG4jAwA2MLgBzDAALAEoGHT1taj04hktoBjNvLwYAXgY6GTSAagYANRgwDGhyUgAOAB47BwA+FjUXHgYqzP8g0JzeBlzMopKyypr7GAam7gwwqPitGfiAX3ml2L15pSYjY1ILGGtax2dXd09UjuDwyNXdGNndRKhk1J7M7LS2vwCLhjhC4tKoOVqgdxs0MN1ev1/kNgaNQZNptd4rc7is4isFkA===).

Same on other places.

Reply:
I have made the suggested change

Reply:
@tannergooding do you wat this to be in the scope of this PR? this PR was about upgrading the APIs mentioned in the description. This is surely a optimization opportunity but just checking if adding this here is fine.

Reply:
It's simply enough to include here. But generally speaking it's nice to separate the two considerations as it helps us more easily track performance differences

Reply:
Ohkay. They are already added.